Mobile Buy SDK/Mobile Buy SDK/Data/BUYClient+Routing.h

```diff
@@ -0,0 +1,57 @@
+//
+//  BUYClient+Routing.h
+//  Mobile Buy SDK
+//
+//  Created by Shopify.
+//  Copyright (c) 2015 Shopify Inc. All rights reserved.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+#import "BUYClient.h"
+
+@interface BUYClient (Routing)
+
+- (NSURL *)routeForAPI;
+- (NSURL *)routeForApps;
+
+- (NSURL *)routeForShop;
```

Comment:
Valid point, but these are only consumed in the client implementation and
are not intended for public consumption

On Tue, May 17, 2016 at 9:56 AM Dima Bart <notifications@github.com> wrote:

> In Mobile Buy SDK/Mobile Buy SDK/Data/BUYClient+Routing.h
> <https://github.com/Shopify/mobile-buy-sdk-ios/pull/165#discussion_r63525818>
> :
>
> > +//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
> > +//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
> > +//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
> > +//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
> > +//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
> > +//  THE SOFTWARE.
> > +//
> > +
> > +#import "BUYClient.h"
> > +
> > +@interface BUYClient (Routing)
> > +
> > +- (NSURL *)routeForAPI;
> > +- (NSURL *)routeForApps;
> > +
> > +- (NSURL *)routeForShop;
>
> I find having the routeFor prefix provides very nice code-completion. It
> basically acts as a name space. Are we sure we don't want to take advantage
> of that?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/Shopify/mobile-buy-sdk-ios/pull/165/files/a482ba44aaad14094a76810e250f2fec73d9f716#r63525818>
>